### PR TITLE
Blocks: Add `of` prop to `Title`

### PR DESCRIPTION
**Pull request by Sidnioulz**
Partially implements #22490

## What I did

* Added the `of` prop for Title, only handling 'meta'
* Added stories
* Updated documentation, using https://github.com/storybookjs/storybook/pull/22552 as an example

## How to test

* Run yarn storybook:ui
* Open Storybook in your browser
* Access Title stories

## Questions for maintainers
* Do I need to update `MIGRATION.md` in this instance?
* Do I need to preserve the default attached behaviour or deprecate it?

## Checklist

- [x] Make sure your changes are tested (stories and/or unit, integration, or end-to-end tests)
- [x] Make sure to add/update documentation regarding your changes
- [x] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "build", "documentation", "maintenance", "dependencies", "other"]`

### CODE REVIEW

File: code/ui/blocks/src/blocks/Title.tsx [hunk 1]
```diff
@@ -3,21 +3,72 @@ import type { FunctionComponent, ReactNode } from 'react';
...
+  /**
+   * Specify content to display as the title.
+   */
   children?: ReactNode;
 }
 
 const STORY_KIND_PATH_SEPARATOR = /\s*\/\s*/;
 
-export const extractTitle = (title: ComponentTitle) => {
+export const getTitleFromContext = (title: ComponentTitle) => {
```
Comment (ndelangen) [inline code/ui/blocks/src/blocks/Title.tsx:24]: Can you check if this is exported outside of the package?
If it is indeed exported, then changing it's name is technically a breaking change.

Comment (Sidnioulz) [reply code/ui/blocks/src/blocks/Title.tsx:24]: Correct, it's used elsewhere, I didn't spot it was exported.

* [ ] Revert to previous name

File: docs/api/doc-block-title.md [hunk 2]
```diff
@@ -24,6 +24,12 @@ import { Title } from '@storybook/blocks';
 
 `Title` is configured with the following props:
 
+### `of`
```
Comment (kylegach) [inline docs/api/doc-block-title.md:27]: Thanks for updating the API reference! ❤️ 

Could you please keep these props in alphabetical order?

File: code/ui/blocks/src/blocks/Title.tsx (hunk 1 above)
Comment (Sidnioulz) [reply code/ui/blocks/src/blocks/Title.tsx]: Reverted!

**Comment by yannbf** [comment]
Hey @JReinhold could you please take a look?

**Comment by ndelangen** [comment]
@JReinhold Is this an easy PR to review & merge?

**Review by JReinhold — APPROVED** [review]
Thank you for the PR and your patience. I modified it slightly, LGTM. ❤️

**Pull request merged**